### PR TITLE
Use historyrequire.HistoryRequire to assert on histories in tests

### PR DESCRIPTION
**Pull request by alexshtin**
## What changed?

Use `historyrequire.HistoryRequire` to assert on histories in tests.

## Why?

To make tests more readable and unify history assertions.

## How did you test it?

Run all tests.

## Potential risks

No risks.

## Documentation

No.

## Is hotfix candidate?

No.

### CODE REVIEW

**Review by stephanos — APPROVED** [review]
I like it; it makes it much clearer what's being asserted.

(Maybe wait for one more Approval before merging)

File: common/testing/historyrequire/history_require.go
Comment (tdeebswihart) [inline common/testing/historyrequire/history_require.go:97]: If this is for testing then please use t.Logf so that it’s easily findable in the testing output

**Review by tdeebswihart — APPROVED** [review]

**Pull request merged**
